File: back/src/main/java/com/back/domain/concert/dto/SeatSelectionResponse.java
package com.back.domain.concert.dto;

import com.back.domain.schedule.entity.ScheduleSeat;
import com.back.domain.schedule.entity.SeatStatus;

import java.util.List;
import java.util.Map;

public record SeatSelectionResponse(
        Long concertId,
        Long scheduleId,
        Map<String, Integer> prices,
        List<SeatDetailResponse> seats
) {
    public static SeatSelectionResponse of(
            Long concertId,
            Long scheduleId,
            Map<String, Integer> prices,
            List<SeatDetailResponse> seats
    ){
        return new SeatSelectionResponse(
                concertId,
                scheduleId,
                prices,
                seats
        );
    }
    public record SeatDetailResponse(
            String seatNumber,
            SeatStatus seatStatus,
            String gradeName
    ) {
        public static SeatDetailResponse from(ScheduleSeat scheduleSeat){
            return new SeatDetailResponse(
                    scheduleSeat.getSeatNumber(),
                    scheduleSeat.getSeatStatus(),
                    scheduleSeat.getGradeName()
            );
        }
    }
}

File: back/src/main/kotlin/com/back/domain/concert/dto/SeatSelectionResponse.kt
package com.back.domain.concert.dto

import com.back.domain.schedule.entity.ScheduleSeat
import com.back.domain.schedule.entity.SeatStatus

data class SeatSelectionResponse(
    val concertId: Long,
    val scheduleId: Long,
    val prices: Map<String, Int>,
    val seats: List<SeatDetailResponse>
) {
    companion object {
        fun of(
            concertId: Long,
            scheduleId: Long,
            prices: Map<String, Int>,
            seats: List<SeatDetailResponse>
        ): SeatSelectionResponse = SeatSelectionResponse(concertId, scheduleId, prices, seats)
    }

    data class SeatDetailResponse(
        val seatNumber: String,
        val seatStatus: SeatStatus,
        val gradeName: String
    ) {
        companion object {
            fun from(scheduleSeat: ScheduleSeat): SeatDetailResponse = SeatDetailResponse(
                scheduleSeat.seatNumber,
                scheduleSeat.seatStatus,
                scheduleSeat.gradeName
            )
        }
    }
}
